tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -46,26 +61,79 @@ private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sende
                 string.Equals(assemblyName.Name, "System.Net.Http", StringComparison.OrdinalIgnoreCase))
             {
+                StartupLogger.Debug("Assembly {0} not found.", args.Name);
                 return null;
             }
 
             var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName.Name}.dll");
 
-            // Only load the main profiler into the default Assembly Load Context.
-            // If Datadog.Trace or other libraries are provided by the NuGet package their loads are handled in the following two ways.
-            // 1) The AssemblyVersion is greater than or equal to the version used by Datadog.Trace, the assembly
-            //    will load successfully and will not invoke this resolve event.
-            // 2) The AssemblyVersion is lower than the version used by Datadog.Trace, the assembly will fail to load
-            //    and invoke this resolve event. It must be loaded in a separate AssemblyLoadContext since the application will only
-            //    load the originally referenced version
-            if (File.Exists(path))
+            if (TryGetCachedAssembly(path, out var cachedAssembly))
             {
+                // The file exists in the Home folder...
+                if (cachedAssembly is not null)
+                {
+                    // The assembly is already loaded.
+                    return cachedAssembly;
+                }
+
+                // Only load the main profiler into the default Assembly Load Context.
+                // If Datadog.Trace or other libraries are provided by the NuGet package their loads are handled in the following two ways.
+                // 1) The AssemblyVersion is greater than or equal to the version used by Datadog.Trace, the assembly
+                //    will load successfully and will not invoke this resolve event.
+                // 2) The AssemblyVersion is lower than the version used by Datadog.Trace, the assembly will fail to load
+                //    and invoke this resolve event. It must be loaded in a separate AssemblyLoadContext since the application will only
+                //    load the originally referenced version
                 StartupLogger.Debug("Loading {0} with DependencyLoadContext.LoadFromAssemblyPath", path);
-                return DependencyLoadContext.LoadFromAssemblyPath(path); // Load unresolved framework and third-party dependencies into a custom Assembly Load Context
+                var assembly = DependencyLoadContext.LoadFromAssemblyPath(path); // Load unresolved framework and third-party dependencies into a custom Assembly Load Context
+                TrySetCachedAssembly(path, assembly);
+                return assembly;
             }
 
+            // The file doesn't exist in the Home folder.
             return null;
         }
+
+        private static bool TryGetCachedAssembly(string path, out Assembly cachedAssembly)
```

Comment:
Not sure about the name, as it implies that if it returns true, then `cachedAssembly` will be non-null, but that's not necessarily the case, it just means the _path_ to assembly is valid. Maybe `IsDatadogAssembly(string path, out Assembly cachedAssembly)` is better, as doesn't imply so strongly that `cachedAssembly` will not be null

Reply:
done in [a554497](https://github.com/DataDog/dd-trace-dotnet/pull/2533/commits/a55449752300a26ae38d963e7cee70e88212c6ca)